Kitodo-DataAccess/src/main/java/org/kitodo/dataaccess/format/xml/XMLWriter.java

```diff
@@ -0,0 +1,337 @@
+            throw new RuntimeException(message != null ? message : e.getClass().getSimpleName(), e);
+        }
+        String qualifiedName = abbr.abbreviateElement(getTypeOrDefault(node));
+        Document document = builder.getDOMImplementation()
+                .createDocument(Namespaces.namespaceOfForXMLFile(getTypeOrDefault(node)), qualifiedName, null);
+        Element root = document.getDocumentElement();
+        if ((node instanceof Node) && (node instanceof IdentifiableNode)) {
+            root.setAttribute(abbr.abbreviateAttribute(getTypeOrDefault(node), RDF.ABOUT),
+                ((IdentifiableNode) node).getIdentifier());
+        }
+        nodeToElement(node, document, abbr, root);
+        for (Entry<String, String> nsAttr : abbr.namespaceSetForXMLFile()) {
+            root.setAttribute(nsAttr.getKey(), nsAttr.getValue());
+        }
+        return document;
+    }
+
+    /**
+     * Transforms this node to a DOM element.
+     *
+     * @param object
+     *            object to convert
+     *
+     * @param document
+     *            DOM document to use to create the element
+     * @param abbr
+     *            a namespace handler initialised with the document root
+     * @return this node as a DOM element
+     */
+    private static Element toElement(ObjectType object, Document document, Namespaces abbr) {
+        String type = object instanceof AccessibleObject ? getTypeOrDefault((AccessibleObject) object)
+                : XLink.HREF.getIdentifier();
+        Element result = document.createElement(abbr.abbreviateElement(type));
+        if (object instanceof Node) {
+            nodeToElement((Node) object, document, abbr, result);
+        } else if (object instanceof Literal) {
+            if (object instanceof LangString) {
+                result.setAttribute(abbr.abbreviateAttribute(type, XML.LANG.getIdentifier()),
+                    ((LangString) object).getLocale().toLanguageTag());
+            }
+            String value = ((Literal) object).getValue();
+            if (Space.requiresPreservation(value)) {
+                result.setAttribute(abbr.abbreviateAttribute(type, XML.SPACE.getIdentifier()),
+                    Space.PRESERVE.getValue());
+            }
+            result.setTextContent(value);
+        }
+        return result;
+    }
+
+    /**
+     * Outputs this node as an XML document to a file.
+     *
+     * @param node
+     *            node to convert
+     *
+     * @param file
+     *            File to write to.
+     * @param indent
+     *            number of spaces of each indent level. A negative number
+     *            disables wrapping
+     * @param namespaces
+     *            an external defined mapping of namespaces to abbreviations
+     * @throws IOException
+     *             if the writing fails
+     */
+    public static void toFile(Node node, File file, int indent, Map<String, String> namespaces) throws IOException {
```

Comment:
Should this method to be used to store data on local filesystem in a process directory?

Reply:
This method stores data in a simple XML file. I do not know yet which data it will store where. (I would prefer to store the data in the process directory in another format, which is more straightforward. But technically, you can.)

Reply:
So, you are adding methods which are maybe used? Why not implement a method when this method is needed?